### PR TITLE
#181414701 ch(Setup unit testing) Setup unit testing with Mocha & chai

### PR DESCRIPTION
**Pull request by Robertishimwe**
### What does this PR do?
 Setup unit testing
### Description of Task to be completed?
To ensure the back end has all functionality working as intended. You may use Mocha as the test runner and `chai` and `chai-http` for the unit tests.

### How should this be manually tested?
Get started By Running commands bellow:
  - clone repo
  - switch to [ch-setup-test-181414759](https://github.com/atlp-rwanda/code-ninjas-be/tree/ch-setup-test-181414701) branch
  - ``yarn install``: to install all dependencies
  - Create .env file
  - Set up .env as stated in .env example
  - run `yarn migrate` to set up table
  -  run `yarn seed` to populate table with sample data
  - run `yarn test` to start unit testing
  
  - After testing completed successfully you will be able to see passed and failed test, and you will be able to see NYC report .

### What are the relevant pivotal tracker stories?
[#181414701](https://www.pivotaltracker.com/story/show/181414701)

### CODE REVIEW

**Comment by ericus123** [comment]
Rebase from develop

**Review by Procech20 — APPROVED** [review]
LGTM

File: package.json [hunk 1]
```diff
@@ -1,50 +1,63 @@
...
+        "migrate": "yarn sequelize-cli db:migrate",
+        "down": "yarn sequelize-cli db:migrate:undo",
+        "seed": "yarn sequelize-cli db:seed:all",
+        "unseed": "yarn sequelize-cli db:seed:undo:all",
+        "build": "babel src -d lib",
+        "start": "yarn run build && node lib/index.js",
+        "dev": "nodemon --exec babel-node -r dotenv/config src/index.js",
+        "lint": "eslint src/*.js --fix",
+        "prepare": "husky install",
+        "test": "c8 --reporter=text --reporter=lcov ./node_modules/.bin/mocha --timeout 10000 --exit --require @babel/register"
```
Comment (ericus123) [inline package.json]: Please make sure you are using the test environment when running tests

**Review by ericus123 — CHANGES_REQUESTED** [review]

**Review by itisWasp — APPROVED** [review]
As I helped to fix the Issues.

File: .github/workflows/Node.js.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        with:
+          codeclimate-test-reporter-id: d1a7faa48cd92b07f5561dce5e948e3e0c49ffb4f4c469bd1a3b12b5ee4a96ff
+          command: after-build
+<<<<<<< HEAD
+=======
+    env:
+      CI: true
+      DB_CONNECT: ${{ secrets.DB_CONNECT }}
+      DEV_DB: ${{ secrets.DEV_DB }}
+>>>>>>> 90321a90a9f3e869738f11c9ef50b725598dd432
```
Comment (ericus123) [inline .github/workflows/Node.js.yml]: Why am I seeing these conflicting changes?
Make sure you resolve these.

File: src/database/config/database.js [hunk 3]
```diff
@@ -2,8 +2,18 @@ import Sequelize from 'sequelize';
...
 
-const dbase = new Sequelize(process.env.DB_CONNECT, {
+if (process.env.NODE_ENV == 'development') {
+  db_url = process.env.DEV_DB;
+}
+if (process.env.NODE_ENV == 'test') {
+  db_url = process.env.TEST_DB;
+} else {
+  db_url = process.env.DB_CONNECT;
+}
```
Comment (ericus123) [inline src/database/config/database.js]: Find a better way to do this without hardcoding the environments as you already have them in NODE_ENV ("development","test","production")

**Review by ericus123 — CHANGES_REQUESTED** [review]

**Review by ericus123 — APPROVED** [review]

**Pull request merged**